File: src/main/resources/static/app/models/positionVm.ts
///<reference path="wageVm.ts"/>
///<reference path="personVm.ts"/>
/**
 * Created by Hassan on 11/18/2016.
 */
class PositionVm {
    public id: number;
    public title: string;
    public wages: Array<WageVm>;
    public personnel: Array<PersonVm>;


    constructor(id: number, title: string, wages: Array<WageVm>, personnel: Array<PersonVm>) {
        this.setup();
        if (id != undefined)this.id = id;
        if (title != undefined)this.title = title;
        if (wages != undefined)this.wages = wages;
        if (personnel != undefined)this.personnel = personnel;
    }

    private setup() {
        this.id = 0;
        this.title = null;
        this.wages = [];
        this.personnel = [];
    }
}

File: src/main/java/hassan/personnel/managment/models/vm/PositionVm.java
package hassan.personnel.managment.models.vm;

import java.util.List;

/**
 * Created by Hassan on 11/21/2016.
 */
public class PositionVm {
    private int id;

    private String title;

    private List<WageVm> wages;

    private List<PersonVm> personnel;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<WageVm> getWages() {
        return wages;
    }

    public void setWages(List<WageVm> wages) {
        this.wages = wages;
    }

    public List<PersonVm> getPersonnel() {
        return personnel;
    }

    public void setPersonnel(List<PersonVm> personnel) {
        this.personnel = personnel;
    }
}
